dashboard/utils.py

```diff
@@ -363,11 +364,21 @@ def count_courses_passed(self):
         Returns:
             int: A number of passed courses.
         """
-        return (
-            self.final_grade_qset.for_course_run_keys(self.edx_course_keys).passed()
-            .values_list('course_run__course__id', flat=True)
-            .distinct().count()
-        )
+        if self.program.financial_aid_availability:
+            courses = Course.objects.filter(program=self.program)
+            return sum([
+                CombinedFinalGrade.objects.filter(user=self.user, course=course).exists()
```

Comment:
I think this should be checking if the `CombinedFinalGrade` is considered passed, right? Maybe we need to add a `passed` field to that model like in `FinalGrade` so you can then filter and do a simple `exists()` or even `filter(course__in=[course for course in courses if course.has_exam]).count()`?

If you do the latter, you could probably then refactor the `else` clause here into a separate function that takes a lsit of `edx_course_keys` as a parameter and pass `[course.edx_course_key for course in course if not course.has_exam]` which would cut the query overhead for financial aid programs from N (# of courses) to 2.

Reply:
I think the idea for CombinedFinalGrade was that you get one only if you passed the course.

Reply:
Meaning CombinedFinalGrade is created only if the learner passed any course_run and has a passing exam grade (>50%)

Reply:
That doesn't appear to be enforced in `update_or_create_combined_final_grade` and I'm not sure we'd want to do that because wouldn't that mean if you didn't pass you never see the grade on your dashboard?

Reply:
yeah, that correct. If you don't pass you just see the edx grade and the exam grade but not the combined grade. That is how it was originally implemented, before `CombinedFinalGrade`. I also confirmed that with @pdpinch when working on `CombinedFinalGrade`

Reply:
In `update_or_create_combined_final_grade`, `get_best_final_grade_for_course` get only passing final grades and `get_best_proctored_exam_grade` gets only passing exam grades.

Reply:
Ah ok, missed that. Thanks for clarifying.

Reply:
I think it would still be worth trying to cut down on the query count though, right? While it might not be too noticeable on the dashboard it'll probably add up on reindexing everything since the queries are executed serially.